src/sensor.h

```diff
@@ -204,7 +205,7 @@ namespace librealsense
 
         virtual void register_option(rs2_option id, std::shared_ptr<option> option);
         virtual bool try_register_option(rs2_option id, std::shared_ptr<option> option);
-        void unregister_option(rs2_option id);
+        virtual void unregister_option(rs2_option id);
```

Comment:
Why?

Reply:
`register_option` is virtual so subclasses can modify behavior and add functionality. They may also need to modify the behavior of unregister operations.
For example, a sensor that should update its options automatically will need to register/unregister the options both to the `options_container` and the `options_updater`.

Reply:
OK, but I don't see a use. Seems like a leftover.

Reply:
And, again -- right now, the options have to be registered. But I think the overall goal is to get general notifications for all options.

Reply:
Registering all synthetic-sensor options. Playback, record, dds sensors still not implemented.